### PR TITLE
wine: Enable on macOS

### PR DESCRIPTION
**Pull request by angerman**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/emulators/wine/base.nix [hunk 1]
```diff
@@ -6,7 +6,8 @@
   buildScript ? null, configureFlags ? ""
 }:
 
-assert stdenv.cc.cc.isGNU or false;
+assert stdenv.isLinux || stdenv.isDarwin;
```
Comment (matthewbauer) [inline pkgs/misc/emulators/wine/base.nix]: I think we can just remove this. The assert is only needed to prevent a potential evaluation failure below.

**Comment by matthewbauer** [comment]
Incredible work!!!

File: pkgs/misc/emulators/wine/base.nix [hunk 2]
```diff
@@ -115,4 +121,6 @@ stdenv.mkDerivation ((lib.optionalAttrs (! isNull buildScript) {
     description = "An Open Source implementation of the Windows API on top of X, OpenGL, and Unix";
     maintainers = with stdenv.lib.maintainers; [ avnik raskin bendlas ];
   };
-})
+  } // lib.optionalAttrs stdenv.isDarwin {
+    hardeningDisable = [ "fortify" ];
```
Comment (Ericson2314) [inline pkgs/misc/emulators/wine/base.nix]: `hardeningDisable` can go up above with a `lib.optional`. This isn't a mass rebuild, right?

File: pkgs/misc/emulators/wine/base.nix (hunk 1 above)
Comment (Ericson2314) [reply pkgs/misc/emulators/wine/base.nix]: yeah do `meta.platforms` instead.

Comment (Ericson2314) [reply pkgs/misc/emulators/wine/base.nix]: platform.unix + platform.windows, or all or something.

**Review by Ericson2314 — CHANGES_REQUESTED** [review]

File: pkgs/misc/emulators/wine/base.nix [hunk 3]
```diff
@@ -16,6 +15,8 @@ stdenv.mkDerivation ((lib.optionalAttrs (! isNull buildScript) {
 }) // rec {
   inherit name src configureFlags;
 
+  hardeningDisable = [ "fortify" ];
```
Comment (veprbl) [inline pkgs/misc/emulators/wine/base.nix]: ```hardeningDisable = lib.optional stdenv.isDarwin "fortify";```

**Comment by Ericson2314** [comment]
I fixed the remaining issues.

**Pull request merged**